tenant/service_onboarding.go

```diff
@@ -6,12 +6,14 @@ import (
 	"github.com/influxdata/influxdb/v2"
 	icontext "github.com/influxdata/influxdb/v2/context"
 	"github.com/influxdata/influxdb/v2/kv"
+	"go.uber.org/zap"
 )
 
 type OnboardService struct {
 	service     *Service
 	authSvc     influxdb.AuthorizationService
 	alwaysAllow bool
+	log         *zap.Logger
```

Comment:
It's a little odd to have a `log` parameter here when there's a whole middleware layer for logging. IMO the middlewares are more confusing than they're worth and could be replaced by option-setters like `WithOnboardingLogger` below. Holding off on the refactor for now 🙂